### PR TITLE
[fs] Fix FS so it overwrites w and w+ file modes

### PR DESCRIPTION
**Pull request by jprestwo**
Signed-off-by: James Prestwood <james.prestwood@intel.com>

### CODE REVIEW

**Comment by jimmy-huang** [comment]
need to do this for readFileSync and readFileAsync functions as well

File: src/zjs_fs.c [hunk 1]
```diff
@@ -851,6 +858,10 @@ static ZJS_DECL_FUNC_ARGS(zjs_fs_write_file, uint8_t async)
         ERR_PRINT("error opening file, error=%d\n", error);
         goto Finished;
     }
+    if (fs_truncate(&fp, 0) != 0) {
+        ERR_PRINT("could not truncate file: %s\n", path);
```
Comment (jimmy-huang) [inline src/zjs_fs.c]: goto Finished?

**Comment by grgustaf** [comment]
+1

**Comment by jimmy-huang** [comment]
+1

**Pull request merged**